BYOB.md

```diff
@@ -0,0 +1,206 @@
+
+ - `slsa-workflow-inputs` contains a JSON object with a list of key-value pairs for the inputs provided by the [TRW to the SSA during initialization ](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L106). We will see shortly how to use these values.
+ - `slsa-layout-file` is a path to a file containing the layout to generate the attestation. We will see shortly how the format for this file.
+ - `slsa-workflow-secretX`, where X from '1' to '15'. These contain the secrets that the TRW [provides to the SRW during invocation](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L118-L120). Unused secrets [should be defined as unused](https://github.com/laurentsimon/byob-doc/blob/main/internal/callback_action/action.yml#L26-L39). 
+
+#### Outputs
+We [declare the same outputs](https://github.com/laurentsimon/byob-doc/blob/main/internal/callback_action/action.yml#L41-L47) as the existing Actions. These outputs are made available to the TRW by the BYOB framework. They [may be returned by the TRW to the PW](https://github.com/laurentsimon/byob-doc/blob/main/.github/workflows/builder_example_slsa3.yml#L61-L67).
+
+#### Invocation of Existing Action
+We [invoke the existing Action](https://github.com/laurentsimon/byob-doc/blob/main/internal/callback_action/action.yml#L57-L65) as a local Action and pass it the inputs by extracting them from the `slsa-workflow-inputs` argument:
+
+```yaml
+uses: ./../__TOOL_CHECKOUT_DIR__
+id: build
+  with:
+    artifact: ${{ fromJson(inputs.slsa-workflow-inputs).artifact }}
+    content: ${{ fromJson(inputs.slsa-workflow-inputs).content }}
+    username: ${{ fromJson(inputs.slsa-workflow-inputs).username }}
+    password: ${{ inputs.slsa-workflow-secret1 }}
+    token: ${{ inputs.slsa-workflow-secret2 || github.token }}
+```
+
+Note that the `./../__TOOL_CHECKOUT_DIR__` is the path where the TRW repository is checked out by the BYOB framework, so it's accessible locally.
+Notice how we populate the token field: If the user has not passed a value to `inputs.slsa-workflow-secret2`, we default to using the GitHub token `github.token`.
+
+#### Generation of Metadata Layout File
+The last thing to do in the TCA is to [generate the metadata file](https://github.com/laurentsimon/byob-doc/blob/main/internal/callback_action/action.yml#L67-L73) to indicate to the BYOB platform which files to attest to, and which attestations to generate. You can ask the platform to generate several attestations, each attestating to one or more artifacts. The snippet below indicates of a single attestation attesting to a single built artifact `my-artifact`. When the BYOB framework generates the attetation, it will add an extension to it, e.g. `.sigstore` or `.intoto.jsonl` depending on the format used.
```

Comment:
```suggestion
The last thing to do in the TCA is to [generate the metadata layout file](https://github.com/laurentsimon/byob-doc/blob/main/internal/callback_action/action.yml#L67-L73) to indicate to the BYOB platform which files to attest to, and which attestations to generate. You can ask the platform to generate several attestations, each attestating to one or more artifacts. The snippet below indicates a single attestation attesting to a single built artifact `my-artifact`. When the BYOB framework generates the attestation, it will add an extension to it, e.g. `.sigstore` or `.intoto.jsonl` depending on the format used.
```